### PR TITLE
Remove Azure disk in-tree storage plugin

### PR DESCRIPTION
**Pull request by andyzhangx**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Remove Azure disk in-tree storage plugin

Since azure file in-tree storage plugin would be removed in k8s 1.28, so we should still keep legacy azure cloud provider

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Remove Azure disk in-tree storage plugin
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
Remove Azure disk in-tree storage plugin
```

/priority important-soon
/sig cloud-provider
/area provider/azure
/sig storage
/triage accepted

### CODE REVIEW

**Comment by dims** [comment]
cmd/* and vendor/* looks good to me!

/approve

File: pkg/volume/csimigration/plugin_manager.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func (pm PluginManager) IsMigrationCompleteForPlugin(pluginName string) bool {
 	case csilibplugins.AzureFileInTreePluginName:
 		return pm.featureGate.Enabled(features.InTreePluginAzureFileUnregister)
 	case csilibplugins.AzureDiskInTreePluginName:
-		return pm.featureGate.Enabled(features.InTreePluginAzureDiskUnregister)
+		return true
```
Comment (xing-yang) [inline pkg/volume/csimigration/plugin_manager.go:70]: I think this one should not be changed yet.

File: test/e2e/storage/in_tree_volumes.go [hunk 2]
```diff
@@ -38,7 +38,6 @@ var testDrivers = []func() storageframework.TestDriver{
 	drivers.InitEmptydirDriver,
 	drivers.InitCinderDriver,
 	drivers.InitVSphereDriver,
-	drivers.InitAzureDiskDriver,
```
Comment (xing-yang) [inline test/e2e/storage/in_tree_volumes.go:41]: @jsafrane Should this one stay?  I see that entries for Cinder and AWS are still here.

File: cmd/cloud-controller-manager/providers.go [hunk 3]
```diff
@@ -25,7 +25,6 @@ package main
 import (
 	// NOTE: Importing all in-tree cloud-providers is not required when
 	// implementing an out-of-tree cloud-provider.
-	_ "k8s.io/legacy-cloud-providers/azure"
```
Comment (jsafrane) [inline cmd/cloud-controller-manager/providers.go:28]: Please keep Azure cloud provider around.

File: cmd/kube-controller-manager/app/plugins_providers.go [hunk 4]
```diff
@@ -26,7 +26,6 @@ import (
 	"k8s.io/kubernetes/pkg/features"
 	"k8s.io/kubernetes/pkg/volume"
 	"k8s.io/kubernetes/pkg/volume/azure_file"
-	"k8s.io/kubernetes/pkg/volume/azuredd"
```
Comment (jsafrane) [inline cmd/kube-controller-manager/app/plugins_providers.go:29]: Please keep Azure cloud provider around.

File: pkg/kubelet/cadvisor/cadvisor_cloudproviders.go [hunk 5]
```diff
@@ -22,6 +22,5 @@ package cadvisor
 import (
 	// Register cloud info providers.
 	// TODO(#68522): Remove this in 1.20+ once the cAdvisor endpoints are removed.
-	_ "github.com/google/cadvisor/utils/cloudinfo/azure"
```
Comment (jsafrane) [inline pkg/kubelet/cadvisor/cadvisor_cloudproviders.go:25]: Please keep Azure cloud provider around.

File: plugin/pkg/admission/storage/persistentvolume/label/admission.go [hunk 6]
```diff
@@ -205,12 +204,6 @@ func (l *persistentVolumeLabel) findVolumeLabels(volume *api.PersistentVolume) (
 			return nil, fmt.Errorf("error querying GCE PD volume %s: %v", volume.Spec.GCEPersistentDisk.PDName, err)
 		}
 		return labels, nil
-	case volume.Spec.AzureDisk != nil:
-		labels, err := l.findAzureDiskLabels(volume)
-		if err != nil {
-			return nil, fmt.Errorf("error querying AzureDisk volume %s: %v", volume.Spec.AzureDisk.DiskName, err)
-		}
-		return labels, nil
```
Comment (jsafrane) [inline plugin/pkg/admission/storage/persistentvolume/label/admission.go:213]: I am somewhat torn about this. PV labeling is not a feature of the volume plugin per se. I would probably keep the labeling working and remove it once the whole Azure cloud provider is removed, together with AzurFile plugin.

File: test/e2e/framework/providers/azure/azure.go [hunk 7]
```diff
@@ -63,28 +63,6 @@ func (p *Provider) DeleteNode(node *v1.Node) error {
 	return errors.New("not implemented yet")
 }
 
-// CreatePD creates a persistent volume
-func (p *Provider) CreatePD(zone string) (string, error) {
```
Comment (jsafrane) [inline test/e2e/framework/providers/azure/azure.go:67]: Please keep it working, as long as we have Azure cloud provider in the tests. Let's remove it with the cloud provider.

File: test/e2e/storage/drivers/in_tree.go [hunk 8]
```diff
@@ -1314,150 +1313,6 @@ func (v *vSphereVolume) DeleteVolume(ctx context.Context) {
...
-
-var _ storageframework.TestDriver = &azureDiskDriver{}
-var _ storageframework.PreprovisionedVolumeTestDriver = &azureDiskDriver{}
-var _ storageframework.InlineVolumeTestDriver = &azureDiskDriver{}
-var _ storageframework.PreprovisionedPVTestDriver = &azureDiskDriver{}
-var _ storageframework.DynamicPVTestDriver = &azureDiskDriver{}
-var _ storageframework.CustomTimeoutsTestDriver = &azureDiskDriver{}
-
-// InitAzureDiskDriver returns azureDiskDriver that implements TestDriver interface
-func InitAzureDiskDriver() storageframework.TestDriver {
```
Comment (jsafrane) [inline test/e2e/storage/drivers/in_tree.go:1334]: Please keep the tests for now, when we still have Azure SDK. When removing the AzureFile + cloud provider (+ azure SDK deps), still keep just the parts that don't require the SDK - the dynamic provisioning with in-tree AzureDisk / AzureFile should work forever and should not require any special deps. Check AWS or Cinder TestDrivers in this file.

**Review by jsafrane — COMMENTED** [review]
I think we should distinguish between removing AzureDisk volume plugin and the whole Azure cloud provider. This PR has parts of both. Let's remove just the volume plugin here and keep the Cloud provider working, e.g. in e2e tests and PV labeler.

File: cmd/kube-controller-manager/app/plugins_providers.go (hunk 4 above)
Comment (andyzhangx) [reply cmd/kube-controller-manager/app/plugins_providers.go:29]: "k8s.io/legacy-cloud-providers/aws" is already removed here, are we sure we want to keep azuredd here?

Comment (andyzhangx) [reply cmd/kube-controller-manager/app/plugins_providers.go:29]: in L66, we want to remove azuredd in `appendAttachableLegacyProviderVolumes`, so we also need to remove the import azuredd here, right?

File: test/e2e/framework/providers/azure/azure.go (hunk 7 above)
Comment (andyzhangx) [reply test/e2e/framework/providers/azure/azure.go]: fixed

File: plugin/pkg/admission/storage/persistentvolume/label/admission.go (hunk 6 above)
Comment (andyzhangx) [reply plugin/pkg/admission/storage/persistentvolume/label/admission.go]: reverted

File: cmd/cloud-controller-manager/providers.go (hunk 3 above)
Comment (andyzhangx) [reply cmd/cloud-controller-manager/providers.go]: reverted

File: test/e2e/storage/in_tree_volumes.go (hunk 2 above)
Comment (andyzhangx) [reply test/e2e/storage/in_tree_volumes.go]: reverted

File: pkg/volume/csimigration/plugin_manager.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/volume/csimigration/plugin_manager.go:70]: reverted

File: cmd/kube-controller-manager/app/plugins_providers.go (hunk 4 above)
Comment (jsafrane) [reply cmd/kube-controller-manager/app/plugins_providers.go:29]: Oh, sorry, please remove `azuredd`

**Pull request merged**